pkg/pluginregistry/registry.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2021-present Open Networking Foundation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pluginregistry
+
+import (
+	"context"
+	"crypto/tls"
+	"fmt"
+	api "github.com/onosproject/onos-api/go/onos/config/admin"
+	"github.com/onosproject/onos-lib-go/pkg/certs"
+	"github.com/onosproject/onos-lib-go/pkg/errors"
+	"github.com/onosproject/onos-lib-go/pkg/grpc/retry"
+	"github.com/onosproject/onos-lib-go/pkg/logging"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"sync"
+)
+
+var log = logging.GetLogger("registry")
+
+// ModelPlugin is a record of information compiled from the configuration model plugin
+type ModelPlugin struct {
```

Comment:
If you look at the current registry, you will see there are some util functions for Read only/ReadWrite paths which is integrated with model registry + some stratum specefic code part of the loading the plugin. Somehow part of my PR I need those util functions (jsonvalues under model registry) but they should be independent of model registry package. I will try to define them somewhere else but we should compare current model registry plugin with the new one and see how they used to be integrated.

Reply:
Yes, the RO/RW paths addition is on my list to do along with the xpath additions that Sean did recently on master. Unclear what to do with the Stratum-specific stuff however.